testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/security/authentication/BasicAuthenticationMechanismPicketboxRemovedTestCase.java

```diff
@@ -0,0 +1,188 @@
+        try (FileOutputStream fos = new FileOutputStream(cliFile)) {
+            IOUtils.copy(BasicAuthenticationMechanismPicketboxRemovedTestCase.class.getResourceAsStream("enable-elytron.cli"), fos);
+        }
+        runBatch(cliFile);
+        cliFile.delete();
+        reload();
+
+        try (CLIWrapper cli = new CLIWrapper(true)) {
+            cli.sendLine("/subsystem=security:remove", false);
+            reload();
+        }
+    }
+
+    /**
+     * Sends command line to CLI.
+     *
+     * @param line specifies the command line.
+     * @param ignoreError if set to false, asserts that handling the line did not result in a
+     *        {@link org.jboss.as.cli.CommandLineException}.
+     *
+     * @return true if the CLI is in a non-error state following handling the line
+     */
+    public boolean sendLine(String line, boolean ignoreError) {
+        consoleOut.reset();
+        if (ignoreError) {
+            commandCtx.handleSafe(line);
+            return commandCtx.getExitCode() == 0;
+        } else {
+            try {
+                commandCtx.handle(line);
+            } catch (CommandLineException e) {
+                StringWriter stackTrace = new StringWriter();
+                e.printStackTrace(new PrintWriter(stackTrace));
+                fail(String.format("Failed to execute line '%s'%n%s", line, stackTrace.toString()));
+            }
+        }
+        return true;
+    }
+
+    private boolean runBatch(File batchFile) throws Exception {
+        sendLine("run-batch --file=\"" + batchFile.getAbsolutePath() + "\" --headers={allow-resource-service-restart=true} -v", false);
+        return new CLIOpResult(ModelNode.fromStream(new ByteArrayInputStream(consoleOut.toByteArray())))
+                .isIsOutcomeSuccess();
+    }
+
+    private void reload() {
+        ModelNode operation = Util.createOperation("reload", null);
+        ServerReload.executeReloadAndWaitForCompletion(modelControllerClient, operation, (int) SECONDS.toMillis(90), TestSuiteEnvironment.getServerAddress(),
+                TestSuiteEnvironment.getServerPort());
+    }
+
+    /**
+     * Test checks if no NullPointerException is thrown after secured EJB is accessed via http call.
+     *
+     * @param url
+     * @throws Exception
+     */
+    @Test
+    @RunAsClient
+    public void test(@ArquillianResource URL url) throws Exception {
+        HttpResponse response;
+        HttpGet httpget = new HttpGet(url.toString() + "ejb-security");
+        String headerValue = java.util.Base64.getEncoder().encodeToString("user:password".getBytes());
+        Assert.assertNotNull(headerValue);
+        httpget.addHeader("Authorization", "Basic " + headerValue);
+        String text;
+
+        try (CloseableHttpClient httpclient = HttpClients.createDefault()){
+            response = httpclient.execute(httpget);
+            text = Utils.getContent(response);
+
+            Assert.assertFalse(text.contains("java.lang.NullPointerException"));
```

Comment:
You need to test for a specific successes condition (i.e. check that you can invoke a secured EJB and return a result). You can't just check the output for a specific exception name and call it done. The server could still be completely broken but throwing a different exception and this test would pass.